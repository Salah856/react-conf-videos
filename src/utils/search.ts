import { compose, toLower, trim } from 'ramda'
import { remove as removeDiacritics } from 'diacritics'

export const cleanQuery = compose(
  removeDiacritics,
  toLower,
  trim
)

export const isFilterEmpty = (filterValue: string = '') => filterValue.trim() === ''
