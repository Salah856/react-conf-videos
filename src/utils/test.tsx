import * as React from 'react'
import { createMockStore } from 'redux-test-utils'
import { MiddlewareAPI } from 'redux'
import { mount, shallow } from 'enzyme'
import { ReactElement } from 'react'
import { MemoryRouter } from 'react-router'

import { Conference, IndexedConferences, Video } from '../domain'

export const mockStore = (state = {}): MiddlewareAPI<any> => createMockStore(state)

const isStore = (store: any) => store.hasOwnProperty('isActionTypeDispatched')

// shallow mount component with a store in context
export const shallowWithStore = (storeP: any, component: ReactElement<{}>) => {
  const store = isStore(storeP) ? storeP : createMockStore(storeP)
  return shallow<any, any>(component, { context: { store } })
}

export const mountWithStore = (storeP: any, component: ReactElement<{}>) => {
  const store = isStore(storeP) ? storeP : createMockStore(storeP)
  return mount<any, any>(component, { context: { store } })
}

export const wrapWithMemoryRouter = (Component: React.ReactElement<any>) => (
  // key within location prop keeps changing, so manually set
  // so snapshorts are consistent
  <MemoryRouter initialEntries={[{key: 'xxxx'}]} >
    {Component}
  </MemoryRouter>
)

/** Mock data */
export const mockVideo = (): Video => ({
  link: 'a link',
  split: '',
  title: 'test title',
  length: '12:34',
  presenter: 'aaa',
  embeddableLink: 'a link',
})

export const mockConference = (): Conference => ({
  date: 'XX/YY/ZZZ',
  title: 'fake conf title',
  website: 'fake url',
  playlist: 'day 1',
  videos: ['aaa', 'bbb', 'ccc']
})

export const mockIndexedConferences = (): IndexedConferences => ({
  ['XXXX']: mockConference()
})

export const mockRouterState = (pathname = '/conference/8424e37df85b9eccbe48e9a55d93845e/react-conf-2018') => ({
  router: {
    location: {
      pathname: '/conference/8424e37df85b9eccbe48e9a55d93845e/react-conf-2018',
      search: '',
      hash: ''
    }
  }
})

export const mockConferencePageSlice = () => ({
  conferencePage: {
    selectedConferenceId: 'XXXX',
    conference: mockConference()
  }
})

export const mockDataState = () => ({
  data: {
    conferences: mockIndexedConferences()
  }
})
