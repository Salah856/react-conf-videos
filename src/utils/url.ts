import { compose, replace, toLower } from 'ramda';

// replaces white space with - and lower case
export const sluggifyUrl = compose(
  replace(/\s/g, '-'),
  toLower
);
