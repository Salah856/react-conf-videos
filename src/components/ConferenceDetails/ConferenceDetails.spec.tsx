import * as React from 'react'
import toJSON from 'enzyme-to-json'

import { shallow } from 'enzyme'

import { mockConference } from 'utils/test'
import { ConferenceDetails } from './ConferenceDetails'

describe('ConferenceDetails', () => {
  it('should render', () => {
    // arrange
    const props = { conference: mockConference() }

    // act
    const comp = shallow(<ConferenceDetails {...props} />)

    // assert
    expect(toJSON(comp)).toMatchSnapshot()
  })

  it('should display correct text of videos when n > 1', () => {
    // arrange
    const props = { conference: mockConference() }
    const expectedText = '3 videos'

    // act
    const comp = shallow(<ConferenceDetails {...props} />)

    // assert
    expect(comp.find('.details').childAt(1).text()).toEqual(expectedText)
  })

  it('should display correct text of videos when n = 1', () => {
    // arrange
    const props = { conference: mockConference() }
    props.conference.videos = ['aaa']
    const expectedText = '1 video'

    // act
    const comp = shallow(<ConferenceDetails {...props} />)

    // assert
    expect(comp.find('.details').childAt(1).text()).toEqual(expectedText)
  })
})
