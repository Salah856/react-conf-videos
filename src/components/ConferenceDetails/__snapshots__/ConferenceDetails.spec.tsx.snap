// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConferenceDetails should render 1`] = `
<div
  className="root"
>
  <h2
    className="title"
  >
    <a
      href="fake url"
    >
      fake conf title
    </a>
  </h2>
  <p
    className="details"
  >
    <span>
      XX/YY/ZZZ
    </span>
    <span>
      3
       
      videos
    </span>
  </p>
</div>
`;
